### ISSUE TITLE
cant run container with 1.10.0-dev using tls

### ISSUE THREAD

**Issue by pandrew** [docker/docker#19148]
```
docker@vb:~$ docker version
Client:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.2
 Git commit:   18fdbdb
 Built:        Wed Jan  6 22:11:51 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.2
 Git commit:   18fdbdb
 Built:        Wed Jan  6 22:11:51 2016
 OS/Arch:      linux/amd64
```

```
docker@vb:~$ docker info
Containers: 9
Images: 1
Server Version: 1.10.0-dev
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 22
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.1.13-boot2docker
Operating System: Boot2Docker 1.9.1 (TCL 6.4.1); master : a109ce9 - Tue Jan  5 05:58:09 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 996.2 MiB
Name: vb
ID: FSWQ:VNBF:NSJB:EGLR:EDTA:IEJD:7VYW:J5OO:XXWJ:66C4:RBUQ:QA6R
Debug mode (server): true
 File Descriptors: 28
 Goroutines: 50
 System Time: 2016-01-06T23:00:31.072711806Z
 EventsListeners: 0
 Init SHA1:
 Init Path: /usr/local/bin/docker
 Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
 provider=virtualbox

```

```
docker@vb:~$ uname -a
Linux vb 4.1.13-boot2docker #1 SMP Tue Jan 5 05:28:28 UTC 2016 x86_64 GNU/Linux
```

```
docker-machine version 0.6.0-dev, build e0570c0
```

Trying to run a container using certificates created with machine, yes the issue might be in machine but hold on. Running `docker run -it ubuntu` using docker-1.10.0-dev client version on osx does not seem to attach so i copied over the client tls directory to vm i created with machine. Using the settings from `machine env vb` and runnig the same command from the vm the results are the same. Also note that i updated DOCKER_CERT_PATH to the new directory. The stable 1.9.1 client version has no issues running the same command on docker 1.10.0-dev daemon on the vm.

From a 1.10.0-dev osx client:
```
time="2016-01-06T22:29:35.928872766Z" level=debug msg="Calling POST /v1.22/containers/create"
time="2016-01-06T22:29:35.928913437Z" level=debug msg="POST /v1.22/containers/create"
time="2016-01-06T22:29:35.929080561Z" level=debug msg="form data: {\"AttachStderr\":true,\"AttachStdin\":true,\"AttachStdout\":true,\"Cmd\":null,\"Domainname\":\"\",\"Entrypoint\":null,\"Env\":[],\"HostConfig\":{\"Binds\":null,\"BlkioDeviceReadBps\":null,\"BlkioDeviceReadIOps\":null,\"BlkioDeviceWriteBps\":null,\"BlkioDeviceWriteIOps\":null,\"BlkioWeight\":0,\"BlkioWeightDevice\":null,\"CapAdd\":null,\"CapDrop\":null,\"CgroupParent\":\"\",\"ConsoleSize\":[0,0],\"ContainerIDFile\":\"\",\"CpuPeriod\":0,\"CpuQuota\":0,\"CpuShares\":0,\"CpusetCpus\":\"\",\"CpusetMems\":\"\",\"Devices\":[],\"Dns\":[],\"DnsOptions\":[],\"DnsSearch\":[],\"ExtraHosts\":null,\"GroupAdd\":null,\"IpcMode\":\"\",\"Isolation\":\"\",\"KernelMemory\":0,\"Links\":null,\"LogConfig\":{\"Config\":{},\"Type\":\"\"},\"Memory\":0,\"MemoryReservation\":0,\"MemorySwap\":0,\"MemorySwappiness\":-1,\"NetworkMode\":\"default\",\"OomKillDisable\":false,\"OomScoreAdj\":0,\"PidMode\":\"\",\"PortBindings\":{},\"Privileged\":false,\"PublishAllPorts\":false,\"ReadonlyRootfs\":false,\"RestartPolicy\":{\"MaximumRetryCount\":0,\"Name\":\"no\"},\"SecurityOpt\":null,\"ShmSize\":0,\"UTSMode\":\"\",\"Ulimits\":null,\"VolumeDriver\":\"\",\"VolumesFrom\":null},\"Hostname\":\"\",\"Image\":\"ubuntu\",\"Labels\":{},\"OnBuild\":null,\"OpenStdin\":true,\"StdinOnce\":true,\"StopSignal\":\"SIGTERM\",\"Tty\":true,\"User\":\"\",\"Volumes\":{},\"WorkingDir\":\"\"}"
time="2016-01-06T22:29:35.937749441Z" level=debug msg="container mounted via layerStore: /mnt/sda1/var/lib/docker/aufs/mnt/58e894cfc033a0762c5dbbfce83c0353f02e7b28a4b09e866b249a48848d19bb"
2016/01/06 22:29:35 http: TLS handshake error from 192.168.99.1:62558: tls: oversized record received with length 21536
time="2016-01-06T22:29:35.943167511Z" level=debug msg="Calling POST /v1.22/containers/ef22b7cebb1263509f89cb0eac3a65591d6e1da730de0c34a799c928ff3cdd70/start"
time="2016-01-06T22:29:35.943208162Z" level=debug msg="POST /v1.22/containers/ef22b7cebb1263509f89cb0eac3a65591d6e1da730de0c34a799c928ff3cdd70/start"
time="2016-01-06T22:29:35.944495250Z" level=debug msg="container mounted via layerStore: /mnt/sda1/var/lib/docker/aufs/mnt/58e894cfc033a0762c5dbbfce83c0353f02e7b28a4b09e866b249a48848d19bb"
time="2016-01-06T22:29:35.944594822Z" level=debug msg="Assigning addresses for endpoint jovial_almeida's interface on network bridge"
time="2016-01-06T22:29:35.944612914Z" level=debug msg="RequestAddress(LocalDefault/172.17.0.0/16, <nil>, map[])"
time="2016-01-06T22:29:35.947400724Z" level=debug msg="Assigning addresses for endpoint jovial_almeida's interface on network bridge"
time="2016-01-06T22:29:35.996399495Z" level=debug msg="Calling POST /v1.22/containers/ef22b7cebb1263509f89cb0eac3a65591d6e1da730de0c34a799c928ff3cdd70/resize"
time="2016-01-06T22:29:35.996440351Z" level=debug msg="POST /v1.22/containers/ef22b7cebb1263509f89cb0eac3a65591d6e1da730de0c34a799c928ff3cdd70/resize?h=71&w=238"
time="2016-01-06T22:29:35.997110535Z" level=debug msg="Calling GET /v1.22/containers/ef22b7cebb1263509f89cb0eac3a65591d6e1da730de0c34a799c928ff3cdd70/json"
time="2016-01-06T22:29:35.997140674Z" level=debug msg="GET /v1.22/containers/ef22b7cebb1263509f89cb0eac3a65591d6e1da730de0c34a799c928ff3cdd70/json"
```

The same command from stable docker-1.9.1 client on osx:
```

time="2016-01-06T22:31:02.222934489Z" level=debug msg="Calling POST /v1.21/containers/create"
time="2016-01-06T22:31:02.222978348Z" level=debug msg="POST /v1.21/containers/create"
time="2016-01-06T22:31:02.223154924Z" level=debug msg="form data: {\"AttachStderr\":true,\"AttachStdin\":true,\"AttachStdout\":true,\"Cmd\":null,\"Domainname\":\"\",\"Entrypoint\":null,\"Env\":[],\"HostConfig\":{\"Binds\":null,\"BlkioWeight\":0,\"CapAdd\":null,\"CapDrop\":null,\"CgroupParent\":\"\",\"ConsoleSize\":[0,0],\"ContainerIDFile\":\"\",\"CpuPeriod\":0,\"CpuQuota\":0,\"CpuShares\":0,\"CpusetCpus\":\"\",\"CpusetMems\":\"\",\"Devices\":[],\"Dns\":[],\"DnsOptions\":[],\"DnsSearch\":[],\"ExtraHosts\":null,\"GroupAdd\":null,\"IpcMode\":\"\",\"KernelMemory\":0,\"Links\":null,\"LogConfig\":{\"Config\":{},\"Type\":\"\"},\"LxcConf\":[],\"Memory\":0,\"MemoryReservation\":0,\"MemorySwap\":0,\"MemorySwappiness\":-1,\"NetworkMode\":\"default\",\"OomKillDisable\":false,\"PidMode\":\"\",\"PortBindings\":{},\"Privileged\":false,\"PublishAllPorts\":false,\"ReadonlyRootfs\":false,\"RestartPolicy\":{\"MaximumRetryCount\":0,\"Name\":\"no\"},\"SecurityOpt\":null,\"UTSMode\":\"\",\"Ulimits\":null,\"VolumeDriver\":\"\",\"VolumesFrom\":null},\"Hostname\":\"\",\"Image\":\"ubuntu\",\"Labels\":{},\"OnBuild\":null,\"OpenStdin\":true,\"StdinOnce\":true,\"StopSignal\":\"SIGTERM\",\"Tty\":true,\"User\":\"\",\"Volumes\":{},\"WorkingDir\":\"\"}"
time="2016-01-06T22:31:02.223195969Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.1, server: 1.10.0-dev)"
time="2016-01-06T22:31:02.231315011Z" level=debug msg="container mounted via layerStore: /mnt/sda1/var/lib/docker/aufs/mnt/e57cb392f0b20a6c1cb7f9b119b819fd1e5cc71e673b93d1e43b906da45eabb4"
time="2016-01-06T22:31:02.253409350Z" level=debug msg="Calling POST /v1.21/containers/1b2382f596a7f427327a29b1b0f50d4a4a3e6a4f9e980e56366fdc4d3135dd71/attach"
time="2016-01-06T22:31:02.253452447Z" level=debug msg="POST /v1.21/containers/1b2382f596a7f427327a29b1b0f50d4a4a3e6a4f9e980e56366fdc4d3135dd71/attach?stderr=1&stdin=1&stdout=1&stream=1"
time="2016-01-06T22:31:02.253480592Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.1, server: 1.10.0-dev)"
time="2016-01-06T22:31:02.253760016Z" level=debug msg="attach: stdin: begin"
time="2016-01-06T22:31:02.253813088Z" level=debug msg="attach: stdout: begin"
time="2016-01-06T22:31:02.253848987Z" level=debug msg="attach: stderr: begin"
time="2016-01-06T22:31:02.269199197Z" level=debug msg="Calling POST /v1.21/containers/1b2382f596a7f427327a29b1b0f50d4a4a3e6a4f9e980e56366fdc4d3135dd71/start"
time="2016-01-06T22:31:02.269248667Z" level=debug msg="POST /v1.21/containers/1b2382f596a7f427327a29b1b0f50d4a4a3e6a4f9e980e56366fdc4d3135dd71/start"
time="2016-01-06T22:31:02.269273528Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.1, server: 1.10.0-dev)"
time="2016-01-06T22:31:02.270764793Z" level=debug msg="container mounted via layerStore: /mnt/sda1/var/lib/docker/aufs/mnt/e57cb392f0b20a6c1cb7f9b119b819fd1e5cc71e673b93d1e43b906da45eabb4"
time="2016-01-06T22:31:02.270870243Z" level=debug msg="Assigning addresses for endpoint distracted_mcnulty's interface on network bridge"
time="2016-01-06T22:31:02.270890682Z" level=debug msg="RequestAddress(LocalDefault/172.17.0.0/16, <nil>, map[])"
time="2016-01-06T22:31:02.273750784Z" level=debug msg="Assigning addresses for endpoint distracted_mcnulty's interface on network bridge"
time="2016-01-06T22:31:02.332911212Z" level=debug msg="Calling POST /v1.21/containers/1b2382f596a7f427327a29b1b0f50d4a4a3e6a4f9e980e56366fdc4d3135dd71/resize"
time="2016-01-06T22:31:02.332950313Z" level=debug msg="POST /v1.21/containers/1b2382f596a7f427327a29b1b0f50d4a4a3e6a4f9e980e56366fdc4d3135dd71/resize?h=71&w=238"
time="2016-01-06T22:31:02.332977991Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.1, server: 1.10.0-dev)"


```
```

docker@vb:~$ curl https://192.168.99.100:2376/images/json --cert vb/vb/cert.pem --key vb/vb/key.pem --cacert vb/vb/c
a.pem
[{"Id":"sha256:1c9b046c28502c1d1267a7cfda7d89d29c9469d66e886cd79b34b9d03ed4d29d","ParentId":"","RepoTags":["ubuntu:latest"],"RepoDigests":null,"Created":1451933235,"Size":187924575,"VirtualSize":187924575,"Labels":null}]

```

**Comment by HackToday** [comment]
it seems that you run 
docker run -it ubuntu 

with any command to pass that

**Comment by HackToday** [comment]
try docker run -it ubuntu bash

**Comment by pandrew** [comment]
Running without command works with 1.9.1

**Comment by HackToday** [comment]
I tried on docker 1.9.1 and 1.10-dev both same

$ docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                       PORTS               NAMES
b8c17289dfa2        ubuntu              "/bin/bash"         6 seconds ago       Exited (127) 5 seconds ago                       kickass_meninsky


I did not think not pass command make sense

**Comment by tiborvass** [comment]
@pandrew i couldn't reproduce this, would you mind providing more information? Do you still see this with 1.10.0-rc1 ?

**Comment by pandrew** [comment]
@tiborvass Issue resolved using docker 1.10.0-rc1 677c593 and docker-machine 0.6.0-dev, build 38b6ae3. Thank you.

**Comment by tiborvass** [comment]
@pandrew thanks you made my day!
